clean_code/bfs_maze.py

```diff
@@ -0,0 +1,92 @@
+from maze import BoardIntegrityError
+
+
+class BFS_search():
```

Comment:
Great job! this is good implementation of BFS. Now think how you can make a interface named `Search` and derive from it in form that implement BFS will be simple (remember SOLID).